odoo/tests/common.py

```diff
@@ -477,3 +478,50 @@ def can_import(module):
         return False
     else:
         return True
+
+class TagsError(Exception):
+    pass
+
+def tagged(*tags):
+    """
+    A decorator to tag TestCase objects
+    Tags are stored in a set that can be accessed from a 'test_tags' attribute
+    Tags must only contains alaphanumeric chars
+    """
+    def tags_decorator(obj):
+        unallowed_chars_re = re.compile('\W')
```

Comment:
Move that constant outside the function.
Maybe you can define a regexp for *valid* tags instead:
```python
TAGS_RE = re.compile(r"^\w+$")

if not TAGS_RE.match(tag):
    raise ...
```
Note that this regexp excludes empty tags, while yours does not... :smile: